code/game/gamemodes/changeling/powers/tiny_prick.dm

```diff
@@ -66,60 +66,44 @@
-	sting_icon = "sting_transform"
-	chemical_cost = 50
-	dna_cost = 3
-	genetic_damage = 100
-	var/datum/dna/selected_dna = null
-
-/datum/action/changeling/sting/transformation/Trigger()
-	var/mob/user = usr
-	var/datum/changeling/changeling = user.mind.changeling
-	if(changeling.chosen_sting)
-		unset_sting(user)
-		return
-	selected_dna = changeling.select_dna("Select the target DNA: ", "Target DNA")
-	if(!selected_dna)
-		return
-	if((NOTRANSSTING in selected_dna.species.species_traits) || selected_dna.species.is_small)
-		to_chat(user, "<span class='warning'>The selected DNA is incompatible with our sting.</span>")
-		return
-	..()
+/datum/action/changeling/sting/false_armblade //Port from TG, credit to them
+	name = "False Armblade Sting"
+	desc = "We silently sting a human, injecting a retrovirus that mutates their arm to temporarily appear as an armblade. Costs 20 chemicals."
+	helptext = "The victim will form an armblade much like a changeling would, except the armblade is dull and useless."
+	button_icon_state = "sting_armblade"
+	sting_icon = "sting_armblade"
+	chemical_cost = 20
+	dna_cost = 2
+	genetic_damage = 15
+	max_genetic_damage = 25
 
-/datum/action/changeling/sting/transformation/can_sting(var/mob/user, var/mob/target)
-	if(!..())
-		return
-	if((HUSK in target.mutations) || (!ishuman(target)))
-		to_chat(user, "<span class='warning'>Our sting appears ineffective against its DNA.</span>")
-		return FALSE
-	if(ishuman(target))
-		var/mob/living/carbon/human/H = target
-		if(NO_DNA in H.dna.species.species_traits)
-			to_chat(user, "<span class='warning'>This won't work on a creature without DNA.</span>")
-			return FALSE
-	return TRUE
+/obj/item/melee/arm_blade/false
+	desc = "A grotesque mass of flesh that used to be your arm. Although it looks dangerous at first, you can tell it's actually quite dull and useless."
+	force = 5 //Basically as strong as a punch
+	fake = TRUE
 
-/datum/action/changeling/sting/transformation/sting_action(var/mob/user, var/mob/target)
-	add_attack_logs(user, target, "Transformation sting (changeling) (new identity is [selected_dna.real_name])")
-	if(issmall(target))
-		to_chat(user, "<span class='notice'>Our genes cry out as we sting [target.name]!</span>")
+/datum/action/changeling/sting/false_armblade/sting_action(var/mob/user, var/mob/living/carbon/target)
+	add_attack_logs(user, target, "Fake armblade sting (changeling)")
 
-	if(iscarbon(target) && (target.status_flags & CANWEAKEN))
-		var/mob/living/carbon/C = target
-		C.do_jitter_animation(500)
+	if(!target.drop_item())
+		to_chat(user, "<span class='warning'>The [target.get_active_hand()] is stuck to [target.p_their()] hand, we cannot grow a false armblade over it!</span>")
+		return
+	..()
 
-	target.visible_message("<span class='danger'>[target] begins to violenty convulse!</span>","<span class='userdanger'>You feel a tiny prick and a begin to uncontrollably convulse!</span>")
+	var/obj/item/melee/arm_blade/false/blade = new(target,1)
```

Comment:
```suggestion
	var/obj/item/melee/arm_blade/false/blade = new(target, TRUE)
```
Also here the `fake` parameter is not filled and thus is null. Either add it here as a param or remove it from the New parameters. Now it will show a message to the target saying they form the blade silently